### ISSUE TITLE
Ternary operator treeshaking drops PURE annotation

### ISSUE THREAD

**Issue by manucorporat** [rollup/rollup#2951]
- Rollup Version: 1.15.6
- Operating System (or Browser): osx and browser
- Node Version: 

### How Do We Reproduce?

Rollup evaluates the constant expression at compiler time and drops the PURE comment inside the selected branch, given the following example, the PURE annonation should be preserved if `BUILD == true`:

```js
const BUILD = true;

export const isSomething = BUILD ? /*@__PURE__*/ (() => {
	return bigSideEffect();
})() : false;
```

Here's a REPL that also reproduces it:

https://rollupjs.org/repl?version=1.15.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwQlVJTEQlMjAlM0QlMjB0cnVlJTNCJTVDbiU1Q25leHBvcnQlMjBjb25zdCUyMGlzU29tZXRoaW5nJTIwJTNEJTIwQlVJTEQlMjAlM0YlMjAlMkYqJTQwX19QVVJFX18qJTJGJTIwKCgpJTIwJTNEJTNFJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjBiaWdTaWRlRWZmZWN0KCklM0IlNUNuJTdEKSgpJTIwJTNBJTIwZmFsc2UlM0IlNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

### Expected Behavior

The `/*@__PURE__*/` annotation should not be dropped:

```js
const isSomething = /*@__PURE__*/ (() => {
	return bigSideEffect();
})();

export { isSomething };
```

### Actual Behaviour

```js
const isSomething = (() => {
	return bigSideEffect();
})();

export { isSomething };
```

This issue was caught in the wild as part of the stencil's runtime for detecting "Constructable Stylesheets" support:
https://github.com/ionic-team/stencil/blob/05d242d5d6ba558896495df3f7b923ca109c70d9/src/client/client-window.ts#L34-L40

**Comment by lukastaegert** [comment]
Fix at #2955

**Comment by lukastaegert** [comment]
Admittedly, the output now looks less nice in many situations but I agree that preserving comments (escpecially but not only annotations) would be better and I would not manually want to scan for comments. Also in some situations there are now additional parentheses. If someone cares for this, we could think about trimming whitespace and eliminating unneeded parentheses at some point in the future.
